osutil/epoll/epoll.go

```diff
@@ -65,10 +70,14 @@ func Open() (*Epoll, error) {
 
 // Close closes the event monitoring descriptor.
 func (e *Epoll) Close() error {
+	if e.fd == -1 {
```

Comment:
I'm not sure of the use case of this struct, so I could be off here. Is there potential for `e.fd` to be read from multiple goroutines? `registeredFdCount` is mutated/read via the `atomic` functions, which made me think it might be possible.

Reply:
Good point. In fact, even with atomic r/w of `e.fd`, there's potential for a panic if `e.Close()` is called simultaneously from multiple threads, thus triggering `close(e.closed)` multiple times. I think more broadly I should use a `RWLock` which is acquired for write by `Close()`, and for read by all other method calls.